sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/sources/RateStreamMicroBatchStream.scala

```diff
@@ -62,6 +62,7 @@ class RateStreamMicroBatchStream(
 
     val metadataLog =
       new HDFSMetadataLog[LongOffset](session.get, checkpointLocation) {
+        // Writer should not close the given stream when overriding methods.
```

Comment:
Sorry, I meant to just move this doc to `HDFSMetadataLog`.
@HeartSaVioR, can you pick this commit and open a new PR for scaladocs at serialize and deserialize, so that we can credit to @xy2953396112 as a co-author of this change?

Reply:
my mistake... Please review again.Thanks.

Reply:
@HyukjinKwon 
OK. Seems like that would be the better approach to achieve with less effort. I'll submit a PR soon.
@xy2953396112 
No worries about the authorship/credit even I would submit a new PR - you'll be recorded as one of co-authors.